javatests/arcs/core/data/TypeUtilsTest.kt

```diff
@@ -139,6 +143,150 @@ class TypeUtilsTest {
     )
   }
 
+  @Test
+  fun mapFieldTypeToInferredType_singletonType() {
+    val schema = Schema(
+      setOf(SchemaName(testField2)),
+      SchemaFields(
+        mapOf(testField to FieldType.Int),
+        emptyMap()
+      ),
+      "43"
+    ).also {
+      SchemaRegistry.register(it)
+    }
```

Comment:
This is done automatically in the Schema constructor

Reply:
I believe that the setup step of these tests clears out the Schema Registry, so we have to re-register the schemas that we wish to use in these tests.

Would like to avoid that but didn't want to remove the clear incase it invalidates a test.

What's your opinion on the best style here?

Reply:
The setUp method will run before the test method does. So yes it will be cleared during setUp, but the registration happens when you construct the Schema in line 187. So it should already be there in line 195.

That is the right pattern for registering and clearing schemas (clear in setUp/tearDown, and register the specific ones that are needed in each individual test), it's just that you don't need to register it twice.

Reply:
Thanks, had lost some context, your comment explained it really well.

I have removed all double registers that I could find (I believe registering the Schemas stored in the companion object is still needed and so have kept them, but all others could be removed).